stringutil/stringutil.go

```diff
@@ -239,38 +239,30 @@ func NumericOnly(s string) string {
 	return string(nRunes)
 }
 
-// ObfuscateString with asterisks in place of all characters except the first 3
-// characters, fewer if string is shorter than 5 characters
-func ObfuscateString(raw string) string {
-	var obscure int
-	switch len(raw) {
-	case 0, 1:
-		obscure = len(raw)
-	case 2, 3, 4:
-		obscure = len(raw) / 2
-	default:
-		// obscure all but the first 3 characters
-		obscure = len(raw) - 3
+// Obfuscate the raw string from the given direction, 'left' or 'right', with the
+// given number of repitions of the obfuscator string.
+func Obfuscate(raw string, number int, direction string, obfuscator string) string {
+	obfuscation := strings.Repeat(obfuscator, number)
+	if len(raw) <= number {
+		return obfuscation
 	}
-	return fmt.Sprintf("%s%s", raw[0:len(raw)-obscure], strings.Repeat("*", obscure))
+	switch direction {
+	case "left":
```

Comment:
let's put left and right in an enum for ease of use

Reply:
:+1: